src/engine/OptionalJoin.cpp

```diff
@@ -92,6 +93,12 @@ string OptionalJoin::getDescriptor() const {
 ProtoResult OptionalJoin::computeResult([[maybe_unused]] bool requestLaziness) {
   LOG(DEBUG) << "OptionalJoin result computation..." << endl;
 
+  // If the right of the RootOperations is a Service, precompute it's sibling's
```

Comment:
```suggestion
  // If the right of the RootOperations is a Service, precompute the result of its sibling.
```
Similar typo (its vs it's ) in the other operations.